extensions/fontawesome/src/index.tsx

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState } from 'react';
+import { Action, ActionPanel, Clipboard, environment, getPreferenceValues, Grid, showHUD } from '@raycast/api';
+import { useFetch } from '@raycast/utils';
+import fetch from 'node-fetch';
+import { ApiResponse, Icon, IconStyle, Preferences } from './types';
+import useServer from './use-server';
+
+const iconQuery = `
+query {
+  release(version: "6.2.0") {
+    icons {
+      id
+      unicode
+      label
+      unicode
+      familyStylesByLicense {
+        free {
+          style
+        }
+      }
+    }
+  },
+}
+`;
+
+const getLibraryType = (icon: Icon, iconStyle: IconStyle): string => {
+  if (icon.familyStylesByLicense.free.length === 0) {
+    return iconStyle;
+  }
+
+  if (icon.familyStylesByLicense.free[0].style === 'brands') {
+    return 'brands';
+  }
+
+  return iconStyle;
+};
+
+function prettyPrintIconStyle(iconStyle: IconStyle) {
+  return {
+    brands: 'Brands',
+    duotone: 'Duotone',
+    light: 'Light',
+    regular: 'Regular',
+    'sharp-solid': 'Sharp Solid',
+    solid: 'Solid',
+    thin: 'Thin',
+  }[iconStyle];
+}
+
+export default function Command() {
+  const port = useServer();
+  const { iconStyle } = getPreferenceValues<Preferences>();
+
+  const getSvgUrl = (icon: Icon, iconStyle: IconStyle, isDark?: boolean): string => {
+    return `http://localhost:${port}/${getLibraryType(icon, iconStyle)}/${icon.id}.svg?dark=${
```

Comment:
How does this work?

Reply:
@dutzi Did you forget to point it to the deployment server? @pernielsentikaer This is supposed to be an intermediary server that handles requests for SVGs. Its pure function is to alter the SVG response to change the color to white, if requested.

Reply:
1. Raycast grid items accept SVG urls and render whatever is served by that URL
2. For an SVG to render in "white" we need to set the SVG's `fill` to "white"
3. So we need a way to return a modified SVG, given its URL
4. The `useServer` hook sets up a local server that serves Font Awesome icons
5. It starts by listening to the first free port available
6. It adds `fill="white"` to the served SVG, if requested